sklearn/tests/test_cross_validation.py

```diff
@@ -1038,3 +1038,59 @@ def test_check_is_partition():
 
     p[0] = 23
     assert_false(cval._check_is_partition(p, 100))
+
+
+def test_subject_independent_folds():
+    """ Check that the function produces equilibrated folds
+        with no subject appearing in two different folds
+    """
+    # Fix the seed for reproducibility 
+    np.random.seed(0)
```

Comment:
To avoid side effect between tests, please avoid seeding the global PRNG instance but instead use:

```python
rng = np.random.RandomState(0)

...
subjects = np.random.randint(0, n_subjects, n_samples)

```